pkg/agent/ippool/controller.go

```diff
@@ -126,10 +126,16 @@ func (c *Controller) Run(workers int, stopCh chan struct{}) {
 	}
 
 	<-stopCh
-	logrus.Infof("(ippool.Run) stopping IPPool controller")
+
+	logrus.Info("(controller.Run) IPPool controller terminated")
 }
 
 func (c *Controller) runWorker() {
 	for c.processNextItem() {
 	}
 }
+
+func (c *Controller) Stop(stopCh chan struct{}) {
+	logrus.Info("(controller.Stop) stopping IPPool controller")
+	close(stopCh)
+}
```

Comment:
In controller `Stop`, you will also close `stopCh` again. It's invalid to close a channel twice.